src/filter/http_context.rs

```diff
@@ -142,7 +142,13 @@ impl Context for Filter {
 
         match op_res {
             Ok(operation) => {
-                if GrpcService::process_grpc_response(operation, resp_size).is_ok() {
+                if GrpcService::process_grpc_response(
+                    operation,
+                    resp_size,
+                    &mut self.response_headers_to_add,
```

Comment:
This is a code smell... this seems to invert the indirection between the state machine possibly progressing and the result of such a progress... Not a huge deal, I don't think this is even a first. "Just" mentioning it... 

The way I think about it: something happens, this has a "result" (i.e. some output) and that output is dealt with. Which in this particular case would be adding the headers to this `Vec`, but the action knows nothing about that `Vec`, instead it merely returns the entries to add to a caller that then knows what to do with them.

Reply:
Right, so you think it'd be preferable to have some kind of `GrpcResult` containing the code and the response headers to let the filter deal with that

Reply:
Never mind for now... we probably would want to go to the drawing board at some point and design this a little more thoroughly

Reply:
I missed this.. I realise we would likely change a lot of things here, but how about this alternative in latest commit?